File: instrumentation/spray-client-1.3.1/src/main/scala/spray/client/InboundWrapper.scala
/*
 *
 *  * Copyright 2020 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.agent.instrumentation.sprayclient

import java.util
import com.newrelic.api.agent.{ExtendedInboundHeaders, HeaderType}
import spray.http.{HttpHeader}

import scala.collection.JavaConversions

class InboundWrapper(headers: List[HttpHeader]) extends ExtendedInboundHeaders {

  def getHeaderType: HeaderType = {
    HeaderType.HTTP
  }

  def getHeader(name: String): String = {
    headers.find(header => header.is(name.toLowerCase)).map(header => header.value).orNull
  }

  override def getHeaders(name: String): util.List[String] = {
    val tmpHeaders = headers.filter(header => header.is(name.toLowerCase)).map(header => header.value)
    if (tmpHeaders.isEmpty) {
      return null
    }
    JavaConversions.seqAsJavaList(tmpHeaders)
  }

}

File: instrumentation/vertx-core-4.5.0/src/main/java/com/nr/vertx/instrumentation/InboundWrapper.java
/*
 *
 *  * Copyright 2024 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.vertx.instrumentation;

import com.newrelic.api.agent.HeaderType;
import com.newrelic.api.agent.InboundHeaders;
import io.vertx.core.http.HttpClientResponse;

public class InboundWrapper implements InboundHeaders {

    private final HttpClientResponse response;

    public InboundWrapper(HttpClientResponse response) {
        this.response = response;
    }

    @Override
    public HeaderType getHeaderType() {
        return HeaderType.HTTP;
    }

    @Override
    public String getHeader(String name) {
        return response.getHeader(name);
    }

}
